src/main/scala/mesosphere/marathon/state/Container.scala

```diff
@@ -138,6 +146,7 @@ object Container {
     import Network._
     val validGeneralContainer = validator[Container] { container =>
       container.volumes is every(VolumeWithMount.validVolumeWithMount(enabledFeatures))
+      container.linuxInfo is optional(LinuxInfo.validLinuxInfo)
```

Comment:
This validation is only done for Apps here, you're missing the pod validation _completely_ (see `PodValidation` class). I checked it locally and you can definitely submit any seccomp combination for pods.